internal/services/containers/kubernetes_cluster_extension_resource.go

```diff
@@ -408,7 +408,7 @@ func (r KubernetesClusterExtensionResource) Delete() sdk.ResourceFunc {
 				return err
 			}
 
-			if err := client.DeleteThenPoll(ctx, *id, extensions.DefaultDeleteOperationOptions()); err != nil {
+			if err := client.DeleteThenPoll(ctx, *id, extensions.DeleteOperationOptions{ForceDelete: pointer.To(true)}); err != nil {
```

Comment:
Please add a comment above this with an explanation on why this option needs to be set for the delete.